src/fde.h

```diff
@@ -84,8 +84,25 @@ class fde
     /// use I/O methods that maintain an internal-to-them buffer
     void useBufferedIo(READ_HANDLER *, WRITE_HANDLER *);
 
-    int read(int fd, char *buf, int len) { return readMethod_(fd, buf, len); }
-    int write(int fd, const char *buf, int len) { return writeMethod_(fd, buf, len); }
+    /// call the read method set by fde::setIo() requesting len more bytes.
```

Comment:
I recommend avoiding (re)documenting what that "read method" API is:

```suggestion
    /// call the read method set by setIo()
```

If you insist on (re)documenting, then please reduce the inaccuracy of the current (re)documentation attempt:

```suggestion
    /// call the read method set by setIo(), requesting to read up to len more bytes.
```

The above suggestions also avoid using Foo inside Foo.